src/XMakeBuildEngine/Construction/ProjectRootElement.cs

```diff
@@ -1785,6 +1779,20 @@ public void Save(Encoding saveEncoding)
 #endif
         }
 
+        private XmlDocument RemoveImplicits()
```

Comment:
If this turns out to be a hot path, then the ProjectRootElement could be made aware if it has implicit elements and skip this call when it doesn't.